app/services/encryption/kms_client.rb

```diff
@@ -26,17 +26,48 @@ def use_kms?(ciphertext)
       FeatureManagement.use_kms? && self.class.looks_like_kms?(ciphertext)
     end
 
+    # chunk plaintext into ~4096 byte chunks, but not less than 1024 bytes in a chunk if chunking
     def encrypt_kms(plaintext)
-      ciphertext_blob = aws_client.encrypt(
+      if plaintext.bytesize > 4096
+        encrypt_in_chunks(plaintext)
+      else
+        KEY_TYPE[:KMS] + encrypt_raw_kms(plaintext)
+      end
+    end
+
+    # :reek:FeatureEnvy
+    def encrypt_in_chunks(plaintext)
+      plain_size = plaintext.bytesize
```

Comment:
Yeah, a touch more here to describe what this code is doing would be helpful I think